lib/popup.js

```diff
@@ -1,106 +1,83 @@
-    return
-  }
-
-  // Setup listener for dropdown
-  networkEl.addEventListener('change', event => {
-    if (!event.target.value) {
-      alert('Please Select One')
-    } else {
-      displayHostname = event.target.value
-      estimateTime()
-    }
-  })
-
-  dateEl.addEventListener('change', event => {
-    if (!event.target.value) {
-      alert('Please Select One')
-    } else {
-      displayDate = event.target.value
-      estimateTime()
+const app = new Vue({
+  el: '#app',
+  data: {
+    dates: {
+      'Today': new Date(new Date().setHours(0, 0, 0, 0)),
+      'Yesterday': new Date(new Date(new Date().setDate(new Date().getDate() - 1)).setHours(0, 0, 0, 0))
+    },
+    date: 'Today',
+    config: {},
+    stats: {},
+    hostname: '',
+    activeURL: '',
+    noData: false
+  },
+  computed: {
+    tracked: function () {
+      return this.config && this.config.blacklist && this.config.blacklist.includes(this.activeURL)
+    },
+    topSites: function () {
+      const dayStats = this.stats[this.dates[this.date]]
+      // We take the five most visited websites, ordered by the time spent on them
+      return dayStats ? Object.keys(dayStats).sort((a, b) => dayStats[a] <= dayStats[b]).slice(0, 5) : []
+    },
+    timeSpent: function () {
+      if (this.dates[this.date]) {
+        if (this.stats[this.dates[this.date]]) {
+          if (this.stats[this.dates[this.date]][this.hostname]) {
+            this.noData = false
+            return formatTime(this.stats[this.dates[this.date]][this.hostname])
+          } else {
+            this.noData = true
+            return `You didn't visited this site ${this.date.toLowerCase()}!`
+          }
+        } else {
+          this.noData = false
+          return 'No data for this day'
+        }
+      } else {
+        this.noData = false
+        return 'Invalid date'
+      }
+    },
+    validDates: function () {
+      return Object.keys(this.dates).filter(date => this.stats && this.stats[this.dates[date]])
     }
-  })
-
-  function estimateTime () {
-    if (typeof results.hostNavigationStats[dates[displayDate]] === 'undefined' ||
-        typeof results.hostNavigationStats[dates[displayDate]][displayHostname] === 'undefined') {
-      timeSpentEl.textContent = 'No data'
-      return
+  },
+  methods: {
+    track: function () {
+      this.config.blacklist.push(this.hostname)
+      this.stats[this.dates.Today][this.hostname] = 1 // "Fake" time to be sure that it will show something after re-rendering
+      chrome.storage.local.set({ hostNavigationStats: this.stats, config: this.config })
     }
-
-    const elapsedTime = results.hostNavigationStats[dates[displayDate]][displayHostname]
-    updateTimer(elapsedTime)
   }
+})
 
-  // Get hostname of current tab
-  chrome.tabs.query({active: true, currentWindow: true}, tabs => {
-    const currentHostname = new URL(tabs[0].url).hostname
+chrome.storage.local.get({ hostNavigationStats: {}, config: {} }, res => {
```

Comment:
You can maybe move these instructions to the [`created` event hook of Vue](https://vuejs.org/v2/guide/instance.html), in order to have everything in the Vue app